### PR TITLE
feat(Carousel): new component

### LINKED ISSUE #234: [Feat]: New Carousel component
**Description**:
The `Carousel `component enables users to navigate through a collection of content items, such as images or cards, within a confined space. This component is ideal for use in galleries, feature highlights, or any interface requiring a compact content browsing experience.

**Requirements**:

1. **Component Implementation:**

Implement the `Carousel `component in TypeScript.
Utilize SCSS with BEM methodology for styling.
Integrate design tokens from @geneui/tokens for consistent colors, spacing, and animation effects.
Support different navigation options (e.g., arrows, dots) and autoplay functionality.
Allow customization of transition effects and item display settings (e.g., number of items per view).

2. **Unit Tests:**

Write unit tests using Jest and React Testing Library.
Ensure test coverage for all props, including navigation types, autoplay, and transition effects.
Test for edge cases such as empty content, unsupported configurations, and boundary conditions (e.g., first and last items).

3. **Storybook Stories:**

Create a Storybook story for the Carousel component.
Showcase different configurations, such as varying navigation styles, autoplay, and item layouts.
Create controls for the props so users can interact with the Carousel component in Storybook.
Document all props of the public interfaces.

4. **Figma Design:**

Follow the design outlined in the provided [Figma link.](https://www.figma.com/design/XmW63z2fXfHBC95eCM36ed/Gene-UI%3A-Components?node-id=3258-2277&p=f&m=dev)

Checklist:

 

- [ ] Component is implemented in TypeScript.
- [ ]  Styling follows SCSS using BEM methodology.
- [ ]  Design tokens from @geneui/tokens are used for colors, spacing, and animations.
- [ ]  Unit tests written and cover different states and edge cases.
- [ ]  Storybook stories created with examples of all props and configuration options.
- [ ]  Figma design matches the component implementation.

### PR DESCRIPTION
**Pull request by arthur-grigoryan-softconstruct**
### Chromatic Build Summary 
**Build URL:** https://www.chromatic.com/build?appId=66daee7859a843dd87bfb3d7&number=768 
**Storybook URL:** https://66daee7859a843dd87bfb3d7-nvlziiujmn.chromatic.com/

### CODE REVIEW

File: src/components/atoms/Button/Button.tsx [hunk 1]
```diff
@@ -71,6 +71,11 @@ interface IButtonProps {
      * This prop should be used to set placement properties for the element relative to its parent using BEM conventions.
      */
     className?: string;
+
```
Comment (GaroGabrielyan) [inline src/components/atoms/Button/Button.tsx]: Please remove unnecessary empty space.

File: src/components/atoms/Button/Button.tsx [hunk 2]
```diff
@@ -71,6 +71,11 @@ interface IButtonProps {
...
+    /**
+     *  Defines a string value that labels the current element.
```
Comment (GaroGabrielyan) [inline src/components/atoms/Button/Button.tsx]: Please improve the description for example.
``An ARIA label for a button provides a short, descriptive text label for screen readers and other assistive technologies to announce when the button has no visible text or the visible text isn't clear enough on its own. ``

File: src/components/atoms/Button/Button.tsx [hunk 3]
```diff
@@ -71,6 +71,11 @@ interface IButtonProps {
...
+     */
+    ariaLabel?: string;
```
Comment (GaroGabrielyan) [inline src/components/atoms/Button/Button.tsx:77]: You need to group this prop in Button.stories.tsx file. For example in ``others``  group

File: src/components/molecules/Carousel/Carousel.scss [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+    }
+
+    &_direction {
+        &_horizontal {
+            #{ $this }__button {
+                top: 50%;
+                transform: translateY(-50%);
+                transform-origin: top center;
+
+                [dir="rtl"] & {
```
Comment (GaroGabrielyan) [inline src/components/molecules/Carousel/Carousel.scss]: remove ``[dir="rtl"]`` and use context in component.
``const { RTL } = context.globals;``

File: src/components/molecules/Carousel/Carousel.scss [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+                padding-block: var(--guit-ref-spacing-2xsmall);
+            }
+        }
+    }
+
+    &__button {
+        position: absolute;
+        width: var(--guit-sem-dimension-width-medium);
+        height: var(--guit-sem-dimension-height-medium);
+        z-index: 10;
```
Comment (GaroGabrielyan) [inline src/components/molecules/Carousel/Carousel.scss]: please ensure that we need position the ``&__button`` and ``&__dots`` with absolute if yes, why we need to set ``z-index: 10`` not ``1``.

File: src/components/molecules/Carousel/Carousel.scss [hunk 6]
```diff
@@ -0,0 +1,144 @@
...
+        }
+
+        &_active {
+            background: var(--guit-sem-color-background-selected-2);
+            cursor: default;
+        }
+    }
+}
+
+.carouselStory {
```
Comment (GaroGabrielyan) [inline src/components/molecules/Carousel/Carousel.scss]: Do not use the ``component.scss`` file for styling the story. Please remove.

File: src/components/molecules/Carousel/Carousel.stories.tsx [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+
+const meta: Meta<typeof Carousel> = {
+    title: "Molecules/Carousel",
+    component: Carousel,
+    argTypes: {
+        className: args({ control: "false", ...propCategory.appearance }),
+        direction: args({ control: "select", ...propCategory.appearance }),
+        withSlideArrows: args({ control: "boolean", ...propCategory.appearance }),
+        withIndicators: args({ control: "boolean", ...propCategory.appearance }),
+        children: args({ control: "false", ...propCategory.appearance })
```
Comment (GaroGabrielyan) [inline src/components/molecules/Carousel/Carousel.stories.tsx]: The group name of children prop is `content`

File: src/components/molecules/Carousel/Carousel.stories.tsx [hunk 8]
```diff
@@ -0,0 +1,94 @@
...
+    },
+    args: {
+        children: textContent,
+        direction: "horizontal",
+        withSlideArrows: true,
+        withIndicators: true
```
Comment (GaroGabrielyan) [inline src/components/molecules/Carousel/Carousel.stories.tsx]: You can remove this 3 props as the default values of this props are the same.
``direction``
``withSlideArrows``
``withIndicators``

File: src/components/molecules/Carousel/Carousel.stories.tsx [hunk 9]
```diff
@@ -0,0 +1,94 @@
...
+    } as ICarouselProps
+};
+
+export default meta;
+
+type Story = StoryObj<ICarouselProps>;
+
+export const Default: Story = {
+    args: {}
```
Comment (GaroGabrielyan) [inline src/components/molecules/Carousel/Carousel.stories.tsx]: Please remove unused ``args``

File: src/components/molecules/Carousel/Carousel.tsx [hunk 10]
```diff
@@ -0,0 +1,135 @@
...
+ */
+const Carousel: FC<ICarouselProps> = ({
+    className,
+    children = [],
+    direction = "horizontal",
+    withSlideArrows = true,
+    withIndicators = true
+}) => {
+    const [selectedIndex, setSelectedIndex] = useState(0);
+    const count = children.length;
```
Comment (GaroGabrielyan) [inline src/components/molecules/Carousel/Carousel.tsx]: Please rename ``count`` to more meaningful name

File: src/components/molecules/Carousel/Carousel.tsx [hunk 11]
```diff
@@ -0,0 +1,135 @@
...
+
+    const swipeCallbacks = useMemo(() => {
+        if (direction === "horizontal") {
+            return { onSlideLeft: onPrevClick, onSlideRight: onNextClick };
+        }
+
+        return { onSlideUp: onPrevClick, onSlideDown: onNextClick };
+    }, [direction]);
+
+    const ref = useSwipe<HTMLDivElement>(swipeCallbacks);
```
Comment (GaroGabrielyan) [inline src/components/molecules/Carousel/Carousel.tsx]: Rename ``ref`` to for example ``crouselRef``.

File: src/components/molecules/Carousel/Carousel.tsx [hunk 12]
```diff
@@ -0,0 +1,135 @@
...
+                            className={classNames("carousel__dot", { carousel__dot_active: id === selectedIndex })}
+                        />
+                    ))}
+                </div>
+            )}
+        </div>
+    );
+};
+
+Carousel.displayName = "Carousel";
```
Comment (GaroGabrielyan) [inline src/components/molecules/Carousel/Carousel.tsx]: Please remove this line. This is old approach.

File: src/components/molecules/Carousel/index.tsx [hunk 13]
```diff
@@ -0,0 +1,2 @@
+export { ICarouselProps, default } from "./Carousel";
```
Comment (GaroGabrielyan) [inline src/components/molecules/Carousel/index.tsx]:
```suggestion
export { ICarouselProps, default as Carousel } from "./Carousel";
```

File: src/hooks/useDeviceInfo/useDeviceInfo.ts [hunk 14]
```diff
@@ -52,7 +52,9 @@ const useDeviceInfo = (): IDeviceInfo => {
         const isAndroid = detectedOS === "Android";
         const isIOS = detectedOS === "iOS";
 
-        const isMobileDevice = isAndroid || isIOS;
+        const isTablet =
+            /ipad|tablet|kindle|silk|playbook|bb10/i.test(userAgent) || (isAndroid && !/mobile/i.test(userAgent));
+        const isMobileDevice = isAndroid || isIOS || isTablet;
```
Comment (GaroGabrielyan) [inline src/hooks/useDeviceInfo/useDeviceInfo.ts]: Please remove the 'Tablet' option to avoid confusion. If a tablet breakpoint is needed in the future, we can add it as a separate device type.

**Review by GaroGabrielyan — CHANGES_REQUESTED** [review]

**Review by GaroGabrielyan — APPROVED** [review]
